account_banking_ach_direct_debit/models/account_payment_order.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2018 Thinkwell Designs <dave@thinkwelldesigns.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from markupsafe import Markup
+
+from odoo import models
+
+
+class AccountPaymentOrder(models.Model):
+    _inherit = "account.payment.order"
+
+    def generate_payment_file(self):
+        """
+        Creates the ACH Direct Debit file by calling
+        generate_ach_file in countinghouse_ach_base
+        """
+        self.ensure_one()
+        if self.payment_method_id.code == "ACH-In":
+            return self.generate_ach_file()
+        return super().generate_payment_file()
+
+    def generated2uploaded(self):
+        """Write 'last debit date' on mandates
+        Set mandates from first to recurring
+        Set oneoff mandates to expired
+        """
+        # I call super() BEFORE updating the sequence_type
+        # from first to recurring, so that the account move
+        # is generated BEFORE, which will allow the split
+        # of the account move per sequence_type
+        res = super().generated2uploaded()
+        Mandate = self.env["account.banking.mandate"]
+        for order in self:
+            to_expire_mandates = first_mandates = all_mandates = Mandate
+            for payment in order.payment_ids.filtered("move_id.mandate_id"):
+                mandate_id = payment.move_id.mandate_id
```

Comment:
```suggestion
                mandate = payment.move_id.mandate_id
```
It's a recordset, `mandate` is better for naming convention

Reply:
Take the mandate from payment.order, that field is added by `https://github.com/OCA/bank-payment/pull/1470`

Reply:
> Take the mandate from payment.order, that field is added by https://github.com/OCA/bank-payment/pull/1470


cool, it was merged after we investigated